src/CreateAccount.js

```diff
@@ -0,0 +1,25 @@
+import { parse } from 'qs';
+import React, { Component } from 'react';
+import { Nav } from 'react-bootstrap';
+
+class CreateAccount extends Component {
+  signUp(e) {
```

Comment:
I think it might be easier to write these nav links using react-router `Link`, so we don't have to do this custom click handling and can also leverage client-side routing.

```jsx
import { parse } from 'qs';
import React from 'react';
import { Link } from 'react-router-dom';
import { Nav } from 'react-bootstrap';

const CreateAccount = () => {
  const { redirect } = parse(window.location.search, { ignoreQueryPrefix: true });
  const baseUrl = "/sign-up"
  const url = redirect ? `${baseUrl}?redirect=${redirect}` : baseUrl;

  return (
    <Nav.Item>
      <Link to={url}>Create Account</Link>
    </Nav.Item>
  );
};

export default CreateAccount;
```

In `App.js`, we'd also have to move `<Navigation />` to be within the `div` under `Router`.